---
title: "RNAseq data process "
author: "Yang Lyu"
date: "`r Sys.Date()`"
output:
  pdf_document:
    keep_tex: true
    #latex_engine: pdflatex
    fig_caption: yes
    highlight: haddock
    number_sections: yes
    latex_engine: xelatex
      
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo =F,message = F,warning = F)
```


```{r}
rm(list=ls())
#login information for the synapse
Email="" # Your email in synapse
Pass="" # Your password in synapse
Path=getwd()
COUNT_path=file.path(Path,"data" ,"RNAseq","Count")
Unused_samples=c("JH-2-084","JH-2-106") # These samples are not inclucded in the analyssi
Reference_path="/Users/lyu.yang/Tools/R/Public_database/Reference/Human_GSEA/V7.5" # Download GMT files from  from https://www.gsea-msigdb.org/gsea/msigdb
  
dir.create(COUNT_path,recursive = T)


```



```{r}
# Function needed
matrix.please<-function(x) {
  m<-as.matrix(x[,-1])
  rownames(m)<-x[,1]
  m
}

PCA_Plot_3<- function (data,Annotation,VAR,Color) {
  # logcountdata row:genes,column: samples
  pca <- prcomp(data) 
  pca_out<-as.data.frame(pca$x)
  df_out<- pca_out %>%tibble::rownames_to_column(var=VAR)  %>% left_join(., Annotation) 
  #df_out<- merge (pca_out,Annotation,by.x=0,by.y=0) 
  
  # label_color<- factor(df_out[,group])
  ggplot(df_out,aes_string(x="PC1",y="PC2")) +geom_point(aes_string(colour = Color))
}

Batch_count_plot=function(logcount,sample_anno){
  
  logcount_2=logcount%>%t() %>%as.data.frame() %>%mutate(Sample_name=rownames(.))
  Level=sample_anno %>%arrange(batch)%>%.$Sample_name
  data_mod = reshape2::melt(logcount_2, id.vars="Sample_name", 
                            measure.vars=rownames(logcount)) %>%left_join(sample_anno)%>%arrange(batch)
  
  data_mod$Sample_name=factor(data_mod$Sample_name,levels =Level )
  #return(logcount_2)
 ggplot(data_mod)+geom_boxplot(aes(x=Sample_name, y=value, color=batch))+xlab("Sample")+ylab("log2(Count+1)")+theme(axis.text.x=element_blank())
  
  
}

Deseq2_Deseq_function_2<- function (Countdata,Coldata) {
  dds_fil <- DESeq2:: DESeqDataSetFromMatrix(countData =Countdata, colData = Coldata,
                                             design = ~Group)

  # keep <- rowSums(DESeq2::counts(dds_fil)) >= 10
  # dds_fil_2  <- dds_fil[keep,]
  dds_fil_Deg<- DESeq2::DESeq(dds_fil)
  return(dds_fil_Deg)

}

Deseq_vst=function (Countdata,Coldata) {
  deseq2Data <- DESeq2:: DESeqDataSetFromMatrix(countData =Countdata, colData = Coldata,
                                             design = ~Group)
  deseq2VST <- DESeq2::vst(deseq2Data)
  
  # Convert the DESeq transformed object to a data frame

  deseq2VST <-   SummarizedExperiment::assay(deseq2VST)
  #deseq2VST <- as.data.frame(deseq2VST)
   
  head(deseq2VST)
  return(deseq2VST)
}

Deseq_norm_counts=function (Countdata,Coldata) {
  deseq2Data <- DESeq2:: DESeqDataSetFromMatrix(countData =Countdata, colData = Coldata,
                                                design = ~Group)
  dds <- DESeq2::estimateSizeFactors(deseq2Data)
  
  # Convert the DESeq transformed object to a data frame
  

  normalized_counts <- DESeq2::counts(dds, normalized=TRUE) %>%as.data.frame()
}


Pathway_enrichment_analysis=function(singscore_name,cutoff) {
  #Data=data.frame(Name=singscore_name) %>%mutate(Split=unlist(strsplit(Name,split="_")))
  split_list=list()
  for (i in 1:length(singscore_name)) {
    Split=unlist(strsplit(singscore_name[i],split="_"))
    split_list[[singscore_name[i]]]=Split
  }
  
  combine=unlist(split_list) %>%gsub("TP53","P53",.)
  common_words=c(c("UP","DN","GOBP","VS","OF","CELLS","CELL","TARGETS","FETAL","DESCARTES","GOMF","ACTIVITY","REACTOME","HP","MODULE","GOCC","RESPONSE","TRANSPORT","REGULATION","COMPLEX","UP.V1","KO","ABNORMAL","IN","PROCESS","CTRL","PBMC","ION","SIGNALING","WITH","BY","WP","KEGG","DN.V1","WT","A","B","C","D","E","PROTEIN","GENES","POS","HALLMARK","VIA","THE","AND","TO","CANCER","TUMOR","PATHWAY","POSITIVE","TREATED","BINDING"),c(1:9))
  
  combine_stat=data.frame(split=combine)%>%subset(!split%in%common_words)%>%  group_by(split) %>% summarise(n=n()) %>% dplyr::filter(!grepl("DAY",split )) %>% arrange(desc(n)) %>%dplyr::filter(n>=cutoff)
  

  
}

Pathway_select_by_keyword=function(singscore_name,keyword){
  Pathway_list=list()
  for (i in 1:length(singscore_name)) {
    Split=unlist(strsplit(singscore_name[i],split="_"))
    Intersect=intersect(Split,keyword)
    if(length(Intersect)>0){
      Pathway_list[[ singscore_name[i]]]=Intersect[1]
      
    }
    
  }
  
  return(Pathway_list)
  }

Singscore_data_plot_2=function(gene_list,expression_data,expression_data_anno,Annotation_File,stat_method){
  Color=c("blue","red")
  Gene_singscore= singcore_Normalized_GSEA(expression_data,gene_list)
  
  Annotation_file=Annotation_File[,c("Group")]  %>%as.data.frame()
  rownames(Annotation_file)=Annotation_File$Sample_name
  p2=GGplot_singscore_boxplot_2(Gene_singscore[[1]],Annotation_File,"Group",Color,"Sample_name",Method=stat_method)
  
  pheatmap_pathway_3 (expression_data_anno,unlist(gene_list),names(gene_list), Annotation_file,"V1")
  list(p2)
}


singcore_GSEA=function (data,genelist, is.na=T) {
  nm=names(genelist)
  plist=list()
  Data_prepare=singscore::rankGenes (data)
  
  for (i in seq_along(genelist)) { 
    scoredf<- singscore::simpleScore (Data_prepare, upSet=unique(genelist[,i]))%>%.[1]
    colnames(scoredf)=paste0(nm[i],"_singscore")
    plist[[i]]=scoredf
  }
  return(plist)
  #output=do.call("cbind",plist)
}

singcore_Normalized_GSEA=function (data,genelist, is.na=T) {
  nm=names(genelist)
  plist=list()
  Data_prepare=singscore::rankGenes (data)
 # n_genes=length(intersect(genelist,rownames(data)))
  
   if( is.list(genelist)) {
    for (i in 1:length(genelist)) { 
      n_genes=length(intersect(unique(genelist[[i]]),rownames(data)))
      if (n_genes>4){
      scoredf<- singscore::simpleScore (Data_prepare, upSet=unique(genelist[[i]]))%>%.[1]
      scoredf$Normalized=scale(scoredf[,1])
      colnames(scoredf)=c(paste0(nm[i],"_singscore") , paste0(nm[i],"_Normalized_singscore" ))
      if(nrow(scoredf)==ncol(data)) {
        plist[[nm[i]]]=scoredf
      }
      }
     }
  }
  if(is.data.frame(genelist))  {
    
  for (i in seq_along(genelist)) { 
    n_genes=length(intersect(unique(genelist[[i]]),rownames(data)))
    if (n_genes>4){
    scoredf<- singscore::simpleScore (Data_prepare, upSet=unique(genelist[,i]))%>%.[1]
    scoredf$Normalized=scale(scoredf[,1])
    colnames(scoredf)=c(paste0(nm[i],"_singscore") , paste0(nm[i],"_Normalized_singscore" ))
    if(nrow(scoredf)==ncol(data)) {
      plist[[nm[i]]]=scoredf
     }
    }
  }
  }
  return(plist)
  #output=do.call("cbind",plist)
}
```


```{r}
library(biomaRt) 
library(dplyr)
library(ggplot2)
```


```{r}

if (file.exists(file.path(COUNT_path,"Protein_coding_genes.csv"))) {
  protein_coding_genes=read.csv(file.path(COUNT_path,"Protein_coding_genes.csv"))
  
} else{

 mart <- useMart(biomart = "ENSEMBL_MART_ENSEMBL", dataset = "hsapiens_gene_ensembl", host = 'www.ensembl.org')
 protein_coding_genes <- biomaRt::getBM(attributes = c("external_gene_name", "chromosome_name","description"), filters = c("biotype"),values= list(c("protein_coding")), mart = mart)
write.csv(protein_coding_genes,file=file.path(COUNT_path,"Protein_coding_genes.csv"))
}



```


# Download raw counts data from synapse

```{r,eval=F}
# No need to run if the data has been downloaded
synapser::synLogin(email=Email,password=Pass)
?synGet
# JH batch1 data
synapser::synGet(entity="syn52663091", downloadFile=T,downloadLocation=COUNT_path)#
# wu batch1 data
synapser::synGet(entity="syn51544799", downloadFile=T,downloadLocation=COUNT_path)
# WU_PDX_batch1 data
synapser::synGet(entity="syn51539337", downloadFile=T,downloadLocation=COUNT_path)
# WU_batch2
synapser::synGet(entity="syn51559031", downloadFile=T,downloadLocation=COUNT_path)
#
synapser::synGet(entity="syn51539269", downloadFile=T,downloadLocation=COUNT_path)
# WU PDX batch2 

synapser::synGet(entity="syn49556298", downloadFile=T,downloadLocation=COUNT_path)

library(synapserutils)

# annotation file download

synapser::synGet(entity='syn52663357' ,downloadFile=T, downloadLocation = COUNT_path) 
```



```{r}
# remove samples from the sample list
Samplemap=readxl::read_excel(file.path(COUNT_path,"Simple_sample_map_10132023.xlsx"))%>%dplyr::filter(batch!="WU_batch3")%>%dplyr::filter(grepl("RNA",assay))%>%dplyr::filter(tissue!="xenograft passage")%>%dplyr::filter(tissue!="cell line")%>%dplyr::filter(!individualID%in%Unused_samples)

unique(Samplemap$individualID)
```


```{r}
# Merge the count data from 6 files

Count_files=list.files(COUNT_path,full.names = T,pattern = "gene_counts")
Count_list=list()
for (i in 1:length(Count_files)){

  filename=Count_files[i]
 Count_data=readr::read_tsv(filename)%>%dplyr::filter(gene_name%in%protein_coding_genes$external_gene_name)%>%dplyr::select(-gene_id)
  if (ncol(Count_data)>2){
     Count_list[[i]]=Count_data%>%group_by(gene_name)%>%summarise_all(mean)%>%as.data.frame()%>%matrix.please()} 
  else{
  Count_list[[i]]=Count_data%>%group_by(gene_name)%>%summarise_all(mean)%>%as.data.frame()%>%matrix.please()%>%as.data.frame()%>%setNames(colnames(Count_data)[2])
  
}


}
```


```{r}
# gene number is diffenect among batches
for( i in 1:length(Count_list)){
  
  print(nrow(Count_list[[i]]))

}
```



```{r}
# missing gene names in JH_batch1_salmon.merged.gene_counts.tsv and salmon.merged.gene_counts-newID-syn21984813.tsv"
setdiff(rownames(Count_list[[2]]),rownames(Count_list[[1]]))
```


```{r}
# Merge data by using common row names 
Count_data=do.call(cbind,Count_list[c(2:5)])%>%.[rownames(Count_list[[6]]),]
Count_data2=do.call(cbind,Count_list[c(1,6)])
Count_data2=cbind(Count_data,Count_data2)
# Remove WU and MN samples from data; 
Count_data2_clean=Count_data2%>%.[,(grepl("JH",colnames(.)))]
colnames(Count_data2_clean)=gsub("\\.","-",colnames(Count_data2_clean))
#remove cell line and xenograft samples
```


```{r}
# Data preparation for Deseq2

Count_sample_name=data.frame(Syn_speicimenID=colnames(Count_data2_clean))%>%
  mutate(specimenID=substr(Syn_speicimenID,1,14),aliquotID=substr(Syn_speicimenID,16,21))
RNA_Samplemap=Samplemap%>%mutate(Syn_speicimenID= paste(specimenID,aliquotID,sep="-")) %>%dplyr::filter(Syn_speicimenID%in%Count_sample_name$Syn_speicimenID)

Rawcounts=Count_data2_clean%>%as.data.frame()%>%.[,RNA_Samplemap$Syn_speicimenID]

Disease=data.frame(tumorType=c("Plexiform Neurofibroma","Cutaneous Neurofibroma", "Diffuse Infiltrating Neurofibroma" ,"Nodular Neurofibroma" ,"Malignant Peripheral Nerve Sheath Tumor"))%>%mutate(Disease=c("PN","cNF","DifNF","NodNF","MPNST"  ))

RNAseq_anno=RNA_Samplemap%>%mutate(batch=ifelse(specimenID=="JH-2-023-91HBD","WU_batch2",batch))%>%mutate(Sample_name=Syn_speicimenID)%>%left_join(.,Disease)
RNAseq_anno$batch=factor(RNAseq_anno$batch,c("JH_batch1" ,"WU_batch1","WU_batch2","WU_PDX_batch1","WU_PDX_batch2"))


MPNST_PN_RNAseq_anno=RNAseq_anno%>%dplyr::filter(Disease%in%c("MPNST","PN"))

Disease_sum=RNAseq_anno%>%group_by(tumorType)%>%summarise(n=n())
```


```{r}
knitr::kable(Disease_sum,caption="Sample number for each disease in RNAseq data")%>%kableExtra::kable_styling()

```

# Batch effect removing

## Raw count data has batch effect

```{r,fig.cap="PCA plot for the raw data ",fig.width=4,fig.height=6}
Tumor_count_log=log2(Rawcounts+1)

PCA_Plot_3  (t(Tumor_count_log),RNAseq_anno,"Sample_name","batch")
```


```{r,fig.cap="Count plot for the raw data ",fig.width=4,fig.height=6}

Batch_count_plot(Tumor_count_log,RNAseq_anno)

```


## Batch effect was removed by Combat
 
```{r}

batches = sapply(as.character(RNAseq_anno$batch), switch, "WU_PDX_batch1"=4,"WU_PDX_batch2"=5, "WU_batch1" = 2, "WU_batch2" = 3,"JH_batch1"=1,USE.NAMES = F)
 
corrected_data = sva::ComBat(Tumor_count_log,batch =RNAseq_anno$batch,par.prior=T, mean.only = F)



```


```{r,fig.cap="PCA plot for the batch-removed gene count data (labelled by  batch) ", fig.width=8,fig.height=4}
PCA_Plot_3  (t(corrected_data),RNAseq_anno,"Sample_name","batch")
```


```{r,fig.cap="PCA plot for the batch-removed gene count data (labelled by disease) ",fig.width=8,fig.height=4}
PCA_Plot_3  (t(corrected_data),RNAseq_anno,"Sample_name","tumorType")
```

```{r,fig.cap="Bar plot for the batch-removed gene count data ",fig.width=8,fig.height=4, eval=T}
Batch_count_plot(corrected_data,RNAseq_anno)
```



# Differential gene expression between MPNST and PN (using batch removed data)


```{r,include=F}
# Deseq2 package for differential expressed genes analysis 
GTAC_Sample_anno=RNAseq_anno %>%mutate(Group=Disease) 
GTAC_Sample_anno$Group=factor(GTAC_Sample_anno$Group,levels=c("PN","cNF","DifNF","NodNF","MPNST"  ))
rownames(GTAC_Sample_anno)=GTAC_Sample_anno$Sample_name
colnames(corrected_data)==GTAC_Sample_anno$Sample_name
corrected_data_counts=2^(corrected_data) %>%round(.,0)
MPNST_vs_NF_Deseq=Deseq2_Deseq_function_2(corrected_data_counts,GTAC_Sample_anno)
DESeq2::resultsNames(MPNST_vs_NF_Deseq)
```


```{r}
# Calculate differential cepxressed genes between MPNST and PN
MPNST_vs_PN_res <- DESeq2::results(MPNST_vs_NF_Deseq, name = "Group_MPNST_vs_PN", alpha = 0.05) %>%as.data.frame()
MPNST_vs_PN_res_q005=MPNST_vs_PN_res %>%dplyr::filter(padj<0.05)
MPNST_vs_PN_res_q005_up=MPNST_vs_PN_res_q005 %>%dplyr::filter(log2FoldChange>0)%>%arrange(desc(log2FoldChange))

MPNST_vs_PN_res_q005_DN=MPNST_vs_PN_res_q005 %>%dplyr::filter(log2FoldChange<0)%>%arrange(log2FoldChange)
#
Expression_vst=Deseq_vst(corrected_data_counts,GTAC_Sample_anno)
Expression_vst_anno=t(Expression_vst) %>%as.data.frame() %>%tibble::rownames_to_column (.,var="Sample_name" ) %>%left_join(.,GTAC_Sample_anno)%>% arrange(Group,Sample_name)
rownames(Expression_vst_anno)=Expression_vst_anno$Sample_name
```


```{r}
# Normalized the count by VST from Deseq2 package

Expression_vst_anno_MPNST_PN=Expression_vst_anno%>%dplyr::filter(Sample_name%in%MPNST_PN_RNAseq_anno$Sample_name)
Expression_vst_MPNST_PN=Expression_vst[,rownames(Expression_vst_anno_MPNST_PN)]
MPNST_vs_PN_res_2=MPNST_vs_PN_res%>%mutate(gene_symbol=rownames(.))

q005_gene=rownames(MPNST_vs_PN_res_q005)
Expression_vst_anno_MPNST_PN_q005=Expression_vst_anno_MPNST_PN[,c("Sample_name",q005_gene)]
Expression_vst_MPNST_PN_q005=Expression_vst_MPNST_PN[q005_gene,]


```


\newpage

```{r}
# list samples in the list
MPNST_PN_sample=MPNST_PN_RNAseq_anno [,c("Sample_name","individualID","tumorType","Disease")]
knitr::kable(MPNST_PN_sample,caption = "PN and MPNST sample list")%>% kableExtra::kable_styling(latex_options = c("scale_down" ))
```

## Top 20 differential expressed genes

```{r}

Top20_up_genes=MPNST_vs_PN_res_q005%>%arrange(desc(log2FoldChange))%>%.[1:20,]
Down20_up_genes=MPNST_vs_PN_res_q005%>%arrange(log2FoldChange)%>%.[1:20,]

knitr::kable(Top20_up_genes[,c(2,5,6)],caption = "Top 20 Upregulated genes (MPNST vs PN) " ) %>%kableExtra::kable_styling()
knitr::kable(Down20_up_genes[,c(2,5,6)],caption = "Top 20 Downregulated genes (MPNST vs PN) " ) %>%kableExtra::kable_styling(latex_options = c("hold_position"))

```



```{r}
Process_data=file.path(Path,"data","RNAseq","Process_data")
dir(Process_data)
save(GTAC_Sample_anno,MPNST_vs_PN_res,Expression_vst,Expression_vst_anno ,file=file.path(Process_data,"MPNST_VS_PN_Deseq2_outut.rdata"))


```

```{r}
knitr::knit_exit()
```



 

